paymentsheet/src/test/java/com/stripe/android/utils/screenshots/PaparazziRule.kt

```diff
@@ -0,0 +1,165 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import app.cash.paparazzi.DeviceConfig
+import app.cash.paparazzi.Paparazzi
+import app.cash.paparazzi.androidHome
+import app.cash.paparazzi.detectEnvironment
+import com.android.ide.common.rendering.api.SessionParams
+import com.stripe.android.ui.core.PaymentsTheme
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+import java.lang.reflect.Field
+import java.lang.reflect.Modifier as ReflectionModifier
+
+class PaparazziRule(
+    vararg configOptions: Array<out PaparazziConfigOption>,
+) : TestRule {
+
+    private val testCases: List<TestCase> = configOptions.toTestCases()
+
+    private val defaultDeviceConfig = createPaparazziDeviceConfig()
+    private val paparazzi = createPaparazzi(defaultDeviceConfig)
+
+    private var description: Description? = null
+
+    init {
+        makePaparazziWorkForApi33()
+    }
+
+    override fun apply(base: Statement, description: Description): Statement {
+        this.description = description
+        return paparazzi.apply(base, description)
+    }
+
+    fun snapshot(
+        padding: PaddingValues = PaddingValues(vertical = 16.dp),
+        content: @Composable () -> Unit,
+    ) {
+        // This is to close the prepare done as part of the apply.
+        // We need symmetric calls to prepare/close.
+        paparazzi.close()
+
+        val description = requireNotNull(description) {
+            "Description in PaparazziRule can't be null"
+        }
+
+        for (testCase in testCases) {
+            testCase.initialize()
+
+            // We need to update the entire Description to prevent Paparazzi from converting the
+            // name to lowercase.
+            val newDescription = Description.createTestDescription(
+                description.className,
+                description.methodName + testCase.name,
+            )
+
+            paparazzi.prepare(newDescription)
+
+            val deviceConfig = testCase.apply(defaultDeviceConfig)
+            if (deviceConfig != defaultDeviceConfig) {
+                paparazzi.unsafeUpdateConfig(deviceConfig)
+            }
+
+            paparazzi.snapshot {
+                PaymentsTheme(colors = PaymentsTheme.getColors(testCase.isDarkTheme)) {
+                    Surface(color = MaterialTheme.colors.surface) {
+                        Box(
+                            contentAlignment = Alignment.Center,
+                            modifier = Modifier.padding(padding),
+                        ) {
+                            content()
+                        }
+                    }
+                }
+            }
+
+            paparazzi.close()
```

Comment:
I think we probably need to do a try/finally for this close.

Reply:
Why?